cpu/stm32_common/periph/uart.c

```diff
@@ -154,6 +154,41 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     return UART_OK;
 }
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+int uart_mode(uart_t uart, uart_data_bits_t data_bits, uart_parity_t parity,
+              uart_stop_bits_t stop_bits)
+{
+    assert(uart < UART_NUMOF);
+
+    if (parity) {
+        if (data_bits == UART_DATA_BITS_8) {
+            data_bits = USART_CR1_M;
+        }
+        else if (data_bits == UART_DATA_BITS_7) {
+            data_bits = UART_DATA_BITS_8;
+        }
+        else if (data_bits == UART_DATA_BITS_6) {
+            data_bits = UART_DATA_BITS_7;
+        }
+    }
+    if ((data_bits & UART_INVALID_MODE) || (parity & UART_INVALID_MODE)) {
+        return UART_NOMODE;
+    }
+
+#ifdef USART_CR1_M1
+    if (dev(uart)->ISR & USART_ISR_TC) {
```

Comment:
This should actually be `if (!(dedev(uart)->ISR & USART_ISR_TC)) {

Reply:
ACK